File: pusher-polling-app/api/app/core/pusher.js
import Pusher from "pusher";

import config from "../config.js";

const pusher = new Pusher({
  appId: config.pusherAppId,
  key: config.pusherKey,
  secret: config.pusherSecret,
  cluster: "us2",
  useTLS: true
});


export default pusher;

File: pusher-chat-app/api/app/services/pusher.py
import pusher

from app.config import settings


class PusherService:
    __pusher_client: pusher.Pusher = None

    def __init__(self):
        self.__pusher_client = pusher_client = pusher.Pusher(
            app_id=settings.PUSHER_APP_ID,
            key=settings.PUSHER_KEY,
            secret=settings.PUSHER_SECRET,
            cluster="us2",
            ssl=True
        )

    def trigger(self, data):
        self.__pusher_client.trigger("chat-messages", "new-message", data)


pusher_service = PusherService()
